modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -29,86 +30,16 @@ class Engine[D, U](stateL: Lens[D, Engine.State]) {
   type EventType = Event[ConcreteTypes]
   type ResultType = EventResult[ConcreteTypes]
   type UserEventType = UserEvent[ConcreteTypes]
+  type HandleType[A] = HandleP[D, EventType, A]
```

Comment:
Should we consider not using type aliases? They tend to make the code less clear sometimes

Reply:
💯 kill ~~all~~ most type aliases

Reply:
It it is true that sometimes they make the code less clear, but I would argue that the opposite is also true. If I were to remove all type aliases, I would have to replace `HandleType[EventType]` by `StateT[IO, D, (EventResult[ConcreteTypes], Stream[IO, Event[ConcreteTypes]]` all over Engine.scala.
I think that the difference resides in the details that the type alias hides. If they are important to understand the code, then hiding them under a type alias makes the code less clear. If they are  not that relevant, then they become a distraction, and hiding them can make easier to understand the code.
For example, every instance of `StateT` in Engine.scala uses the same state type and the IO monad. The concrete state type is not very relevant, because the access to its attributes is done with Lenses, and IO is only relevant when related to Streams.

Reply:
i guess it could be done with intermetidate types like
`StateT[IO, D, Results]` where results is a class with a member `(EventResult[ConcreteTypes], Stream[IO, Event[ConcreteTypes]])`

I don't know, perhaps all that should be in a separate PR though